kagami/routes/deps.py

```diff
@@ -1,8 +1,24 @@
+from contextlib import asynccontextmanager
 from typing import Annotated
 
 from fastapi import Depends
+from sqlalchemy.ext.asyncio import AsyncSession
 
 from ..core import Supervisor
+from ..database.engine import DatabaseEngine
 from ..server import supervisor
 
 SupervisorDeps = Annotated[Supervisor, Depends(lambda: supervisor)]
+
+
+@asynccontextmanager
+async def some_async_generator():
+    session: AsyncSession
+    session = DatabaseEngine.session_factory
```

Comment:
This type hint can be wrote like:
session: AsyncSession = DatabaseEngine.session_factory()
Note: session_factory() is a function, so you have to call it.

Reply:
Another thing is you can add the return type hint for a function, like:
async def session_generator() -> AsyncSession | None, means this function return AsyncSession or None if Error occurs.